Automattic/wp-calypso
# Plans 2023: Updated billing descriptions for WPCOM introductory offers

Related to https://github.com/Automattic/dotcom-forge/issues/4570
Branched from https://github.com/Automattic/wp-calypso/pull/84645

## Proposed Changes

Updates billing descriptions to cover introductory offers for both month and year intervals and monthly/yearly/biennial/triennial plans. 

## Media

### Signup

<img width="700" alt="Screenshot 2023-12-06 at 4 45 01 PM" src="https://github.com/Automattic/wp-calypso/assets/1705499/16461b57-a869-4b72-a295-a319896fd185">

### Admin

<img width="700" alt="Screenshot 2023-12-06 at 4 45 22 PM" src="https://github.com/Automattic/wp-calypso/assets/1705499/32c22fce-e62c-474f-a2a4-9fa127469058">


## Testing Instructions



Setup:
* Apply D130783-code on sandbox and follow instructions to get intro offers served on `/plans/[site]` and `/start/plans`
* Set USD as currency in SA

Test:
* Ensure `/plans/[ woo site ]` renders $1 introductory offers and billing descriptions correctly
* Ensure `/plans/[ free | paid ]` and `/start/plans` renders the introductory offers and billing descriptions for Personal and Premium plans
* Ensure the rest of the plans in the grids render the correct billing descriptions (including prices and badges being properly displayed and aligned)
* Offers should only be shown for yearly plans. Try any other term `/plans/[site]/monthly`, `/plans/[site]/2-yearly`, `/plans/[site]/3-yearly` and confirm nothing broken

## Pre-merge Checklist
